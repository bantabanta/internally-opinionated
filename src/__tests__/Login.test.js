import { render, screen, fireEvent } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import LogIn from "../components/LogIn";

const mockStore = configureStore([]);

describe("LogIn", () => {
  let store, component;

  beforeEach(() => {
    store = mockStore({
      authedUser: null,
      users: {
        mtsamis: {
          id: "mtsamis",
          name: "Mike Tsamis",
        },
        sarahedo: {
          id: "sarahedo",
          name: "Sarah Edo",
        },
      },
    });
  });

  it("should show correct dropdown default text", () => {
    component = render(
      <Provider store={store}>
        <Router>
          <LogIn />
        </Router>
      </Provider>
    );
    expect(
      screen.getByRole("option", { name: "Select a User..." }).selected
    ).toBe(true);
  });

  it("should register a user selection", () => {
    component = render(
      <Provider store={store}>
        <Router>
          <LogIn />
        </Router>
      </Provider>
    );

    const combobox = screen.getByRole("combobox");
    const selectedUser = screen.getByRole("option", { name: "Sarah Edo" });
    fireEvent.change(combobox, {
      target: {
        value: selectedUser.value,
      },
    });

    expect(selectedUser.selected).toBe(true);
  });

  it("should match the snapshot", () => {
    component = render(
      <Provider store={store}>
        <Router>
          <LogIn />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
