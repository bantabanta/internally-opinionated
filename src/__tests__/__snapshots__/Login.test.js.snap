// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogIn should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="content-wrapper center"
      >
        <div
          class="header"
        >
          <h1
            class="text-standout"
          >
            Internally Opinionated
          </h1>
          <p>
            Ask colleagues questions and vote on popular polls.
          </p>
        </div>
        <div
          class="card log-in"
        >
          <form
            class=""
          >
            <h3>
              Sign In to your account
            </h3>
            <br />
            <select
              style="width: 200px;"
            >
              <option>
                Select a User...
              </option>
              <option
                value="mtsamis"
              >
                Mike Tsamis
              </option>
              <option
                value="sarahedo"
              >
                Sarah Edo
              </option>
            </select>
            <br />
            <button
              disabled=""
              style="width: 205px;"
              type="submit"
            >
              Log In
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="content-wrapper center"
    >
      <div
        class="header"
      >
        <h1
          class="text-standout"
        >
          Internally Opinionated
        </h1>
        <p>
          Ask colleagues questions and vote on popular polls.
        </p>
      </div>
      <div
        class="card log-in"
      >
        <form
          class=""
        >
          <h3>
            Sign In to your account
          </h3>
          <br />
          <select
            style="width: 200px;"
          >
            <option>
              Select a User...
            </option>
            <option
              value="mtsamis"
            >
              Mike Tsamis
            </option>
            <option
              value="sarahedo"
            >
              Sarah Edo
            </option>
          </select>
          <br />
          <button
            disabled=""
            style="width: 205px;"
            type="submit"
          >
            Log In
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
